### PR TITLE
Add 'replay-name' property to event bridge Event

### PR DESCRIPTION
**Pull request by ThomasAribart**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://aws.amazon.com/fr/blogs/compute/archiving-and-replaying-events-with-amazon-eventbridge/

### CODE REVIEW

**Comment by ThomasAribart** [comment]
Hi, sorry I didn't take the time to do all the checks.

**Review by jamesorlakin — APPROVED** [review]
LGTM, thank you! The DefinitelyTyped build appears to be wobbly (union types appearing in a different order on TS 4.7?) but that's unrelated to this change.

Possibly consider adding a test to `eventbridge-tests.ts:38`, though as it's an optional property and needs approval from a DT maintainer anyway I'm not too worried about it:
```ts
"replay-name": 'test-replay-event-name',
```

**Review by jamesorlakin — APPROVED** [review]
Thanks @ThomasAribart :+1:

File: types/aws-lambda/trigger/eventbridge.d.ts
Comment (jakebailey) [inline types/aws-lambda/trigger/eventbridge.d.ts:18]: Skimming some other types in this package, I suspect this should have `| undefined` to support `exactOptionalPropertyTypes`.

Comment (simonbuchan) [reply types/aws-lambda/trigger/eventbridge.d.ts:18]: My understanding is those were automatically converted by MS when they added the option to avoid breaking everyone's existing code (with the option enabled). Otherwise it would never be usable.

I don't think that applies to new types.

Comment (simonbuchan) [reply types/aws-lambda/trigger/eventbridge.d.ts:18]: Strictly, it should be `| undefined` if it can, in fact, be present but undefined, which is not true for JSON-parsed input values and is for JSON-stringified output values.

Comment (jakebailey) [reply types/aws-lambda/trigger/eventbridge.d.ts:18]: I think we have been pushing for contributions to ensure that things work (@peterblazejewicz definitely has), but we are in sore need for a little more documentation on this in our README as a gotcha.

Comment (peterblazejewicz) [reply types/aws-lambda/trigger/eventbridge.d.ts:18]: yes, we promote optional usage (not enforced) to provide better interop in ecosystem (for projects using strict features consuming DT definitions) - where it make sense (user input data/configurations/options/React props)

Comment (jamesorlakin) [reply types/aws-lambda/trigger/eventbridge.d.ts:18]: To clarify, are all new optional types proposed in DT definitions expected to be in the form `| undefined` as well as `?`, or only if there's a difference in runtime behaviour between `undefined` and 'not present whatsoever' in a given package?

Comment (jakebailey) [reply types/aws-lambda/trigger/eventbridge.d.ts:18]: If there's a difference in behavior between explicitly undefined and "not present", then you wouldn't want to leave the `?`. If there's no difference, then having both makes things more compatible for those who do enable the flag.

It's not that this is a strict requirement for DT packages; there have certainly been PRs merged that don't make the distinction, and it's not that hard to fix if someone reports that they can't use the types.

I try my best to mention it when I see that the package already has `| undefined`, solely for consistency, but overall we need better reference for this.

Comment (jamesorlakin) [reply types/aws-lambda/trigger/eventbridge.d.ts:18]: Aah yep that makes sense, I'll advise adding `| undefined` as appropriate in future aws-lambda PRs but won't block them. Thanks!

Comment (simonbuchan) [reply types/aws-lambda/trigger/eventbridge.d.ts:18]: For this package in particular, events should not have it, so that `in` guards work and type machinery is simpler, and results should, so that conditionally present properties can be much more nicely defined. But as a rule of thumb, sure, you're much more likely to need it present than you are to need it not present.

**Pull request merged**
